### PR TITLE
Windows support

### PR DESCRIPTION
**Pull request by miiiiiiich**
## Summary

agent-walker now runs on Windows. The dashboard reads logs under `%USERPROFILE%`
(`.claude\projects`, `.codex\sessions`, `.gemini\antigravity-cli`), and we ship
a prebuilt Windows binary alongside the existing macOS / Linux ones via the
same `npx agent-walker` / `bunx agent-walker` install path. WSL keeps working
unchanged.

## Changes

- New `src/paths.rs` built on `dirs::home_dir` — home / cache / downloads
  resolution lives in one place, no part of the codebase reads `HOME` directly
  anymore.
- The five sites that used to read `HOME` (default log dirs, cache, share-card
  save path, project label, Claude project name) all route through it.
- `project_from_cwd` strips the home prefix case-insensitively on Windows
  (matching the case-insensitive filesystem), requires a path-component
  boundary so `C:\Users\metadata` doesn't get clipped against home
  `C:\Users\me`, and uses `cwd.get(..home.len())` so a multi-byte UTF-8 cwd
  cannot panic.
- `dist-workspace.toml` builds `x86_64-pc-windows-msvc`; `.github/workflows/ci.yml`
  adds `windows-latest` to the test matrix, so regressions surface on PRs.
- README (EN + JA), CHANGELOG, and Cargo bumped to 0.3.0.

## Test

- `cargo clippy --all-targets -- -D warnings` clean.
- `cargo test` — 43 passing (two new regression tests for `strip_home_prefix`,
  one Unix-gated and one Windows-gated via `cfg!(windows)`).
- Cross-model review through `sk:codex-cli` — first pass flagged a
  case-sensitivity bug and a multi-byte panic, both fixed; the third pass
  returned `fix verified`.
- True Windows behavior is exercised by the new `windows-latest` CI job below.

### CODE REVIEW

File: src/collector/mod.rs [hunk 1]
```diff
@@ -23,13 +23,41 @@ use crate::model::{Collection, DurationEvent, ScanStats, SessionTouch, ToolEvent
...
+/// `C:\Users\metadata` does not get stripped against home `C:\Users\me`.
+#[cfg(windows)]
+fn strip_home_prefix(cwd: &str, home: &str) -> Option<String> {
+    // `get` keeps us safe if `home.len()` lands inside a multi-byte UTF-8
+    // character in `cwd`; `split_at` would panic there.
+    let head = cwd.get(..home.len())?;
+    let rest = cwd.get(home.len()..)?;
+    if !head.eq_ignore_ascii_case(home) {
+        return None;
+    }
```
[the remark this answers is not in the document]
Comment (miiiiiiich) [reply src/collector/mod.rs:73]: Fixed.

**Change**: the Windows branch of `strip_home_prefix` now normalizes `\` to `/` on both `head` and `home` before the case-insensitive compare, so a forward-slashed cwd (typical from npm/Node tooling) still matches a backslashed home.
**File**: `src/collector/mod.rs:42-60`

File: src/paths.rs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    Ok(home_dir()?.join(".cache").join("agent-walker"))
+}
+
+/// Default save directory for the share card. Prefers `<home>/Downloads` when
+/// it exists, otherwise the home directory itself.
+pub fn downloads_dir() -> Result<PathBuf> {
+    let home = home_dir()?;
+    let downloads = home.join("Downloads");
+    Ok(if downloads.is_dir() { downloads } else { home })
+}
```
[the remark this answers is not in the document]
Comment (miiiiiiich) [reply src/paths.rs:33]: Fixed.

**Change**: `paths::downloads_dir` now consults `dirs::download_dir()` first (Known Folders / NSDownloadsDirectory / XDG) and falls back to the home directory only when the OS doesn't report one.
**File**: `src/paths.rs:25-33`

File: src/collector/mod.rs [hunk 3]
```diff
@@ -381,6 +407,40 @@ mod tests {
...
+            assert_eq!(
+                strip_home_prefix(r"c:\users\me\code\app", r"C:\Users\me"),
+                Some(r"code\app".to_owned()),
+            );
+            // Non-boundary sibling does not strip.
+            assert_eq!(
+                strip_home_prefix(r"C:\Users\metadata", r"C:\Users\me"),
+                None
+            );
+        }
```
[the remark this answers is not in the document]
Comment (miiiiiiich) [reply src/collector/mod.rs]: Fixed.

**Change**: the Windows test now covers the mixed-slash case (`C:/Users/me/code/app` vs home `C:\\Users\\me`) plus the existing exact-match and lowercase-drive cases. The Unix test grew the same non-boundary case.
**File**: `src/collector/mod.rs:411-451`

File: src/collector/mod.rs [hunk 4]
```diff
@@ -23,13 +23,41 @@ use crate::model::{Collection, DurationEvent, ScanStats, SessionTouch, ToolEvent
...
+    if !rest.is_empty() && !rest.starts_with(['/', '\\']) {
+        return None;
+    }
+    Some(rest.trim_start_matches(['/', '\\']).to_owned())
+}
+
+#[cfg(not(windows))]
+fn strip_home_prefix(cwd: &str, home: &str) -> Option<String> {
+    cwd.strip_prefix(home)
+        .map(|rest| rest.trim_start_matches(['/', '\\']).to_owned())
```
[the remark this answers is not in the document]
Comment (miiiiiiich) [reply src/collector/mod.rs]: Fixed.

**Change**: the Unix branch now requires the same path-component boundary as the Windows branch — `/Users/metadata/app` and `/Users/me-work/app` against home `/Users/me` return None and fall back to the full path instead of producing bogus project labels.
**File**: `src/collector/mod.rs:62-73`

File: src/collector/mod.rs [hunk 5]
```diff
@@ -205,11 +233,9 @@ struct CacheEntry {
...
 
 fn cache_path(cache_name: &str) -> Option<PathBuf> {
-    let home = std::env::var_os("HOME")?;
     Some(
-        PathBuf::from(home)
-            .join(".cache")
-            .join("agent-walker")
+        crate::paths::cache_dir()
+            .ok()?
             .join(format!("{cache_name}-v{CACHE_VERSION}.bin")),
```
[the remark this answers is not in the document]
Comment (miiiiiiich) [reply src/collector/mod.rs:278]: Fixed.

**Change**: `store_cache` now removes the existing cache file before `fs::rename`, so Windows runs can overwrite the previous cache instead of silently failing. Unix POSIX rename overwrites atomically, so the extra unlink there is a no-op on a missing path.
**File**: `src/collector/mod.rs:279-289`

**Comment by miiiiiiich** [comment]
@codex review

File: src/collector/mod.rs [hunk 6]
```diff
@@ -252,6 +290,13 @@ fn store_cache(cache_name: &str, cache: &CacheFile) {
...
     let temp = path.with_extension("tmp");
     if fs::write(&temp, bytes).is_ok() {
+        // `fs::rename` on Windows refuses to overwrite an existing
+        // destination, so the cache file would never advance and every active
+        // user would reparse every log on every run. Remove the stale cache
+        // first (failure is harmless — the rename below reports the real
+        // outcome). Unix POSIX rename already overwrites atomically; the
+        // extra unlink there is a no-op on a missing path.
+        let _ = fs::remove_file(&path);
         let _ = fs::rename(&temp, &path);
```
[the remark this answers is not in the document]
Comment (miiiiiiich) [reply src/collector/mod.rs]: Fixed.

**Change**: removed the `fs::remove_file` call — `std::fs::rename` overwrites atomically on Unix and uses `MoveFileExW + MOVEFILE_REPLACE_EXISTING` on Windows, so the explicit unlink was breaking the Unix atomicity guarantee for no Windows benefit. (My previous fix here was based on a stale assumption I did not verify; thanks for catching it.)
**File**: `src/collector/mod.rs:278-289`

File: src/collector/mod.rs [hunk 7]
```diff
@@ -23,13 +23,53 @@ use crate::model::{Collection, DurationEvent, ScanStats, SessionTouch, ToolEvent
...
+/// match normalizes both to `/` before comparing — npm/Node tooling often
+/// records cwds with forward slashes even on Windows. Requires a path-
+/// component boundary after the prefix so that `C:\Users\metadata` does not
+/// get stripped against home `C:\Users\me`.
+#[cfg(windows)]
+fn strip_home_prefix(cwd: &str, home: &str) -> Option<String> {
+    // `get` keeps us safe if `home.len()` lands inside a multi-byte UTF-8
+    // character in `cwd`; `split_at` would panic there.
+    let head = cwd.get(..home.len())?;
+    let rest = cwd.get(home.len()..)?;
```
[the remark this answers is not in the document]
Comment (miiiiiiich) [reply src/collector/mod.rs:67]: Fixed.

**Change**: the Windows branch now `trim_end_matches(['/', '\\'])` on `home` (and bails when the trimmed string is empty) so a drive-root home like `D:\\` doesn't make `home.len()` swallow the separator.
**File**: `src/collector/mod.rs:53-76`

File: src/collector/mod.rs [hunk 8]
```diff
@@ -23,13 +23,53 @@ use crate::model::{Collection, DurationEvent, ScanStats, SessionTouch, ToolEvent
...
+    }
+    Some(rest.trim_start_matches(['/', '\\']).to_owned())
+}
+
+/// Same shape on Unix: require a path-component boundary so that home
+/// `/Users/me` does not silently strip a cwd like `/Users/metadata/app` into
+/// `tadata/app` (an attribution bug the previous `{home}/` prefix avoided).
+#[cfg(not(windows))]
+fn strip_home_prefix(cwd: &str, home: &str) -> Option<String> {
+    let rest = cwd.strip_prefix(home)?;
```
[the remark this answers is not in the document]
Comment (miiiiiiich) [reply src/collector/mod.rs:95]: Fixed.

**Change**: the Unix branch trims a trailing `/` on `home` for the same reason, so `/home/me/` still strips cleanly.
**File**: `src/collector/mod.rs:78-94`

File: src/collector/claude.rs [hunk 9]
```diff
@@ -64,8 +64,17 @@ fn project_label(path: &Path, root: &Path) -> Option<String> {
...
+    // is a relative project path. Exact on macOS / Linux. Windows-native
+    // Claude Code's flattening rule (especially what happens to the `:` after
+    // the drive letter) hasn't been confirmed, so when the flattened prefix
+    // doesn't match we just fall back to the leading-`-` trim — the cwd field
+    // in the session is preferred as the project label whenever present.
+    let home_prefix = crate::paths::home_dir()
+        .ok()
+        .and_then(|home| home.to_str().map(ToOwned::to_owned))
+        .map(|home| format!("{}-", home.replace(['/', '\\'], "-")))
         .unwrap_or_default();
```
[the remark this answers is not in the document]
Comment (miiiiiiich) [reply src/collector/claude.rs:84]: Fixed.

**Change**: `normalize_project_name` trims a trailing separator before flattening, so a home such as `/home/me/` or `D:\\` no longer yields a double-hyphen prefix.
**File**: `src/collector/claude.rs:66-86`

File: src/collector/mod.rs [hunk 10]
```diff
@@ -23,13 +23,53 @@ use crate::model::{Collection, DurationEvent, ScanStats, SessionTouch, ToolEvent
...
 pub fn project_from_cwd(cwd: &str) -> String {
-    let stripped = std::env::var("HOME")
+    let home = crate::paths::home_dir()
         .ok()
-        .and_then(|home| cwd.strip_prefix(&format!("{home}/")).map(ToOwned::to_owned))
-        .unwrap_or_else(|| cwd.to_owned());
-    stripped.trim_start_matches('/').to_owned()
+        .and_then(|home| home.to_str().map(ToOwned::to_owned));
+    home.and_then(|home| strip_home_prefix(cwd, &home))
+        .unwrap_or_else(|| cwd.trim_start_matches(['/', '\\']).to_owned())
```
[the remark this answers is not in the document]
Comment (miiiiiiich) [reply src/collector/mod.rs]: Fixed.

**Change**: `project_from_cwd` substitutes `"~"` when stripping leaves an empty label, so sessions whose cwd is exactly the home directory render as `~` in the PROJECTS panel instead of a blank row.
**File**: `src/collector/mod.rs:25-42`

File: src/collector/claude.rs [hunk 11]
```diff
@@ -64,8 +64,17 @@ fn project_label(path: &Path, root: &Path) -> Option<String> {
...
+    // every path separator replaced by `-`. Strip the home prefix so the label
+    // is a relative project path. Exact on macOS / Linux. Windows-native
+    // Claude Code's flattening rule (especially what happens to the `:` after
+    // the drive letter) hasn't been confirmed, so when the flattened prefix
+    // doesn't match we just fall back to the leading-`-` trim — the cwd field
+    // in the session is preferred as the project label whenever present.
+    let home_prefix = crate::paths::home_dir()
+        .ok()
+        .and_then(|home| home.to_str().map(ToOwned::to_owned))
+        .map(|home| format!("{}-", home.replace(['/', '\\'], "-")))
```
[the remark this answers is not in the document]
Comment (miiiiiiich) [reply src/collector/claude.rs]: Fixed.

**Change**: the Claude home-prefix flattener now also replaces `:` (drive separators) alongside the path separators, so a home of `C:\\Users\\me` produces `C-Users-me-` instead of the un-matchable `C:-Users-me-`. Windows-native Claude flattening still isn't independently confirmed, but this aligns the prefix with how an actual Windows path component can be formed.
**File**: `src/collector/claude.rs:66-86`

File: src/collector/mod.rs [hunk 12]
```diff
@@ -23,13 +23,53 @@ use crate::model::{Collection, DurationEvent, ScanStats, SessionTouch, ToolEvent
 const CACHE_VERSION: u32 = 7;
 
 /// Normalize a working-directory path into a project label: strip the home
-/// prefix, keep the real path separators ("/Users/me/code/app" -> "code/app").
+/// prefix, keep the real path separators ("/Users/me/code/app" -> "code/app",
+/// "C:\\Users\\me\\code\\app" -> "code\\app").
```
[the remark this answers is not in the document]
Comment (miiiiiiich) [reply src/collector/mod.rs]: Deferred for a follow-up — tracked locally.

**Reason**: the common-prefix collapse currently assumes `/`-delimited labels. Teaching it to recognise `\\` as well is worthwhile but I'd rather pair the change with a real Windows-native Claude trace so the splitting matches the actual directory shape we end up displaying. Holding off until I have that input.

**Comment by miiiiiiich** [comment]
@codex review

File: src/collector/mod.rs [hunk 13]
```diff
@@ -252,6 +290,13 @@ fn store_cache(cache_name: &str, cache: &CacheFile) {
...
     if fs::write(&temp, bytes).is_ok() {
+        // `fs::rename` on Windows refuses to overwrite an existing
+        // destination, so the cache file would never advance and every active
+        // user would reparse every log on every run. Remove the stale cache
+        // first (failure is harmless — the rename below reports the real
+        // outcome). Unix POSIX rename already overwrites atomically; the
+        // extra unlink there is a no-op on a missing path.
+        let _ = fs::remove_file(&path);
         let _ = fs::rename(&temp, &path);
     }
```
[the remark this answers is not in the document]
Comment (miiiiiiich) [reply src/collector/mod.rs]: Fixed.

**Change**: `fs::remove_file` was removed entirely in `ce6a8f7` (same conclusion you reached here — `std::fs::rename` is already atomic on Unix and uses `MOVEFILE_REPLACE_EXISTING` on Windows, so the unlink was hurting Unix readers for no Windows benefit).
**File**: `src/collector/mod.rs:278-289`

File: src/collector/mod.rs (hunk 10 above)
[the remark this answers is not in the document]
Comment (miiiiiiich) [reply src/collector/mod.rs]: Fixed.

**Change**: `project_from_cwd` now substitutes `"~"` when the strip leaves an empty label (`ce6a8f7`), so sessions whose cwd is exactly the home directory render as `~` instead of a blank PROJECTS row.
**File**: `src/collector/mod.rs:25-42`

File: src/collector/mod.rs [hunk 14]
```diff
@@ -23,13 +23,53 @@ use crate::model::{Collection, DurationEvent, ScanStats, SessionTouch, ToolEvent
...
+    if !head
+        .replace('\\', "/")
+        .eq_ignore_ascii_case(&home.replace('\\', "/"))
+    {
+        return None;
+    }
+    if !rest.is_empty() && !rest.starts_with(['/', '\\']) {
+        return None;
+    }
+    Some(rest.trim_start_matches(['/', '\\']).to_owned())
```
[the remark this answers is not in the document]
Comment (miiiiiiich) [reply src/collector/mod.rs]: Fixed.

**Change**: `strip_home_prefix` (Windows branch) now normalizes the remainder to forward slashes (`e05d512`), so the same repository visited as `C:\\Users\\me\\code\\app` and `C:/Users/me/code/app` collapses to one project key (`code/app`).
**File**: `src/collector/mod.rs:53-80`

File: src/collector/claude.rs (hunk 11 above)
[the remark this answers is not in the document]
Comment (miiiiiiich) [reply src/collector/claude.rs]: Fixed.

**Change**: the Claude home-prefix flattener replaces `:` alongside the path separators (`ce6a8f7`), so a Windows home `C:\\Users\\me` produces a matchable `C-Users-me-` instead of the un-matchable `C:-Users-me-`.
**File**: `src/collector/claude.rs:66-86`
